packages/itwinui-react/src/core/utils/components/ShadowRoot.tsx

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+const isBrowser = typeof document !== 'undefined';
```

Comment:
nit: Maybe this can be reused in other places. E.g. `useIsomorphicLayout`

Reply:
I doubt it. `useIsClient` already exists and is better for 99% of the cases.

Utilities/helpers should generally be colocated where they are used, until they are needed in multiple places.

Reply:
I mean, useIsomorphicLayout uses `typeof window !== 'undefined'`, which I think is similar to `typeof document !== 'undefined'`.

https://github.com/iTwin/iTwinUI/blob/ba8714532461d29ba42829321f4098a1fc4fc41a/packages/itwinui-react/src/core/utils/hooks/useIsomorphicLayoutEffect.tsx#L15

Reply:
To reiterate the `nit` part, this is non-blocking. How it is is also alright 👍. This is me just leaving nit-pick type comments, but how it is is also alright.

Reply:
Going to leave it as-is for now. It's a simple boolean and requires the same (or more) number of characters to import it vs redeclare it.